src/daft-plan/src/source_info.rs

```diff
@@ -1,24 +1,24 @@
 use daft_core::schema::SchemaRef;
 
 pub enum SourceInfo {
-    FilesInfo(FilesInfo),
+    ParquetFilesInfo(ParquetFilesInfo),
```

Comment:
Until there's deviation in what data we're holding in the `SourceInfo` structs across Parquet, CSV, JSON, etc. file types, I think that we should keep it as a `FilesInfo` enum variant containing a `FilesInfo` struct that contains an additional `FileFormat` enum:
```rust
pub enum SourceInfo {
    FilesInfo(FilesInfo),
}

impl SourceInfo {
    pub fn schema(&self) -> SchemaRef {
        use SourceInfo::*;
        match self {
            FilesInfo(files_info) => files_info.schema.clone(),
        }
    }
}

pub enum FileFormat {
    Parquet,
    Csv,
    Json,
}

pub struct FilesInfo {
    pub filepaths: Vec<String>, // TODO: pull in some sort of URL crate for this
    pub file_format: FileFormat,
    pub schema: SchemaRef,
}
```